src/DynamoCore/Services/StabilityTracking.cs

```diff
@@ -174,12 +174,13 @@ public static void Startup()
         }
 
         /// <summary>
-        /// To check whether the last shutdown is clean(no crash)
+        /// To check whether the last shutdown is clean(no crash or unexpected shutdown happens)
         /// </summary>
         private static bool IsLastShutdownClean()
         {
             var ret = Registry.GetValue(REG_KEY, SHUTDOWN_TYPE_NAME, CRASHING_SHUTDOWN_VALUE) as string;
-            if (null != ret && string.CompareOrdinal(ret, CRASHING_SHUTDOWN_VALUE) == 0)
+            if (null != ret && (string.CompareOrdinal(ret, CRASHING_SHUTDOWN_VALUE) == 0
+                || string.CompareOrdinal(ret, ASSUMING_CRASHING_SHUTDOWN_VALUE) == 0))
```

Comment:
Good point! But I guess these are compatible. Or if not, we can build utility functions which are compatible.